src/components/common/PrivateRoute.js

```diff
@@ -0,0 +1,31 @@
+import React from "react"
+import { Route, withRouter, Redirect } from "react-router-dom"
+import { connect } from "react-redux"
+
+import { isEmpty } from '../../actions/account'
+
+const PrivateRoute = ({component: Component, ...rest}) => (
+   <Route 
+      {...rest} 
+      render={(props) => (
+         isEmpty().payload
```

Comment:
@marcinbodnar I'm not sure I understand the way `isEmpty` is used. Looks like here it's the one from `import { isEmpty } from '../../actions/account'` and so doesn't get dispatched. What is the intended flow of using `isEmpty` as action?

Reply:
@vgrichina I'm not sure if it is the best variant, but the idea was not to duplicate wallet class object and at the same time check directly if there is an account available. In the future, we can also easily add request status or global alert to it.

Reply:
@marcinbodnar If this isn't dispatching Redux action, it shouldn't be a Redux action. If the goal is just to have wallet singleton, I assume we can solve this by providing it using context? https://reactjs.org/docs/context.html

Also I assume this can be solved if `isEmpty` is part of Redux state which gets updated as end result of dispatching `handleRefreshAccount`?